app/src/main/java/org/mozilla/fenix/collections/CollectionCreationView.kt

```diff
@@ -27,7 +27,7 @@ import org.mozilla.fenix.R
 import org.mozilla.fenix.components.metrics.Event
```

Comment:
(moving this here to allow a comment chain)

> Any reason to not use [WebURLFinder](https://searchfox.org/mozilla-mobile/source/android-components/components/support/utils/src/main/java/mozilla/components/support/utils/WebURLFinder.kt) or modify it work for your use case?

@jonalmeida I wasn't aware of it.  I should have asked you guys if AC had anything that might help.

I was also trying to match existing behavior in desktop and FFTV as best I could, which meant copying some existing code (and rewriting JS).

I think in the long run here, it might be worth investing in a Rust component that handles these questions in a canonical way, and can be shared between all Moz products.  This is my _I'm-way-behind-schedule-but-we-need-something-now_ attempt.  I'm not particularly proud of it, but I'm trying.

Reply:
I peeked at the PR, after the title caught my interest, so I was curious if we had considered if there was something that didn't work with the WebURLFinder. Refactoring is always possible in the future (and not a one time thing!).

Really sorry, I didn't mean to sound like the effort wasn't worth it, you're doing great!

Reply:
I don't think you did!  I think the correct approach here would be to write a rock-solid component that can be shared between desktop and mobile, because otherwise we're duplicating _very_ error-prone code.  I'm frustrated with my code, not with your comment.  Very sorry to have implied otherwise!